src/OpenTK.Windowing.GraphicsLibraryFramework/Input/KeyboardState.cs

```diff
@@ -8,58 +8,54 @@
 //
 
 using System;
+using System.Collections;
 using System.Diagnostics;
 using System.Text;
+using OpenTK.Windowing.Common;
 
-namespace OpenTK.Windowing.Common.Input
+namespace OpenTK.Windowing.GraphicsLibraryFramework
 {
     /// <summary>
     /// Encapsulates the state of a Keyboard device.
     /// </summary>
-    public struct KeyboardState : IEquatable<KeyboardState>
+    public sealed class KeyboardState
     {
-        // Allocate enough ints to store all keyboard keys
-        private const int IntSize = 32;
+	    // These arrays will mostly be empty since the last integer used is 384. That's only 48 bytes though.
+        private BitArray _keys = new BitArray((int)Keys.LastKey);
+        private BitArray _keysPrevious = new BitArray((int)Keys.LastKey);
 
-        private const int NumInts = ((int)Key.LastKey / IntSize) + 1;
+        private static readonly Keys[] KeyValues = (Keys[])Enum.GetValues(typeof(Keys));
```

Comment:
This is only ever used in the `ToString` function, there is no need to keep a member.